### PR TITLE
Don't pass null pointers to LAPACK, even during workspace queries.

### LINKED ISSUE #6132: Failures with latest MKL
Intel just released the 16.0.3 Parallel studio, and I now have a bunch of test failures of the form:
```
======================================================================
ERROR: test_decomp_update.TestQRinsert_D.test_Mx1_economic_p_col
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/larsoner/custombuilds/nose/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/larsoner/.local/lib/python2.7/site-packages/scipy/linalg/tests/test_decomp_update.py", line 881, in test_Mx1_economic_p_col
    q1, r1 = qr_insert(q, r, u, col, 'col', overwrite_qru=False)
  File "scipy/linalg/_decomp_update.pyx", line 1816, in scipy.linalg._decomp_update.qr_insert (scipy/linalg/_decomp_update.c:28082)
  File "scipy/linalg/_decomp_update.pyx", line 2132, in scipy.linalg._decomp_update.qr_insert_col (scipy/linalg/_decomp_update.c:31692)
ValueError: The 7th argument to ?geqrf was invalid
```
I read in their release notes that they made some checks more strict, this must be the effect.

### PR DESCRIPTION
**Pull request by ewmoore**
Fixes #6132

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Preliminary result is good:
```
larsoner@bunk:~/custombuilds/scipy$ python runtests.py -v -t scipy/linalg/tests/test_decomp_update.py:TestQRinsert_D.test_Mx1_economic_p_col
Building, see build.log...
Build OK
Running unit tests for scipy
NumPy version 1.12.0.dev0+1fc180b
NumPy relaxed strides checking option: True
NumPy is installed in /home/larsoner/.local/lib/python2.7/site-packages/numpy-1.12.0.dev0+1fc180b-py2.7-linux-x86_64.egg/numpy
SciPy version 0.18.0.dev0+9aa2256
SciPy is installed in /home/larsoner/custombuilds/scipy/build/testenv/lib/python2.7/site-packages/scipy
Python version 2.7.11+ (default, Apr 17 2016, 14:00:29) [GCC 5.3.1 20160413]
nose version 1.3.7
test_decomp_update.TestQRinsert_D.test_Mx1_economic_p_col ... ok

----------------------------------------------------------------------
Ran 1 test in 0.028s

OK
```
Let me build the whole thing from scratch just to be sure.

File: scipy/linalg/_decomp_update.pyx.in [hunk 1]
```diff
@@ -725,9 +724,15 @@ cdef int qr_block_col_insert(int m, int n, blas_t* q, int* qs,
         # First zero the lower part of the new columns using a qr.
 
         # query the workspace,
+        # set tau to point at something to keep new MKL working.
+        tau = &c
         info = geqrf(m-n+p, p, index2(r, rs, n-p, k), m, tau, &c, -1)
+        if info < 0:
+            return libc.stdlib.abs(info)
```
Comment (Eric89GXL) [inline scipy/linalg/_decomp_update.pyx.in]: Why return the abs?

Comment (ewmoore) [inline scipy/linalg/_decomp_update.pyx.in]: It is so that the code that prints the error message knows if the error was in geqrf or ormqr.  See line 2016.

Comment (Eric89GXL) [inline scipy/linalg/_decomp_update.pyx.in]: Ahh thanks

**Comment by Eric89GXL** [comment]
Tests now pass, +1 for merge. Thanks for the quick fix.

**Comment by cgohlke** [comment]
+1
This fixes 52 test errors with Intel MKL 11.3.3 on Windows.

**Pull request merged**